### PR TITLE
Replace ActiveJob queue adapter with shoryuken (instead of sidekiq)

### PR DESCRIPTION
**Pull request by sunil-sadasivan**
Part 3 of #3182. 

At this point, the shoryuken gem is installed and shoryuken workers are running on the environments.  This turns SQS as the queue backend for ActiveJobs. 

**To Test**
```
$ rails c
> HeartbeatTasksJob.perform_later
```
 - an outstanding message should end up in an sqs queue (visible [here](https://console.amazonaws-us-gov.com/sqs/home?region=us-gov-west-1#view-messages:selected=https://sqs.us-gov-west-1.amazonaws.com/008577686731/development_low_priority;noRefresh=true;prefix=$))

- Run the shoryuken worker
```
$ shoryuken start -q development_high_priority development_low_priority -R
```

### CODE REVIEW

File: config/application.rb [hunk 1]
```diff
@@ -47,5 +47,11 @@ class Application < Rails::Application
     # eFolder API URL to retrieve appeal documents
     config.efolder_url = ENV["EFOLDER_EXPRESS_URL"]
     config.efolder_key = ENV["EFOLDER_API_KEY"]
+    config.active_job.queue_adapter = :shoryuken
+
+    # sqs details
+    config.active_job.queue_name_prefix = ENV['DEPLOY_ENV'] || Rails.env
```
Comment (sunil-sadasivan) [inline config/application.rb]: @joofsh wanted to point you here.  Unsure if creating a helper function and importing the helper in the `config/application.rb` is ideal.

File: config/initializers/shoryuken.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# set up default exponential backoff parameters
+ActiveJob::QueueAdapters::ShoryukenAdapter::JobWrapper
+  .shoryuken_options(retry_intervals: [3.seconds, 30.seconds, 5.minutes, 30.minutes, 2.hours, 5.hours])
+
+Shoryuken.configure_server do |config|
+  # placeholder for shoryuken server middleware
```
Comment (joofsh) [inline config/initializers/shoryuken.rb]: It looks like this PR actually turns on use of SQS in production. Will we initially lose the sidekiq middleware? I think that will break a few things such as reporting of job errors to Sentry

File: config/application.rb (hunk 1 above)
Comment (joofsh) [reply config/application.rb]: Would it be possible to set the DEPLOY_ENV if not set? `ENV['DEPLOY_ENV'] ||= Rails.env` at the top of this file. And then we can just reference it here.

If you do want to abstract it, I think you can put it in the ApplicationController and successfully reference it here like [this](https://github.com/department-of-veterans-affairs/caseflow/blob/master/app/controllers/application_controller.rb#L208-L212)

File: config/initializers/shoryuken.rb (hunk 2 above)
Comment (sunil-sadasivan) [reply config/initializers/shoryuken.rb]: ah yeah good one @joofsh, let's chat about this.  The middleware interface for the shoryuken server is different than sidekiq :\.  Would love to bounce off the approach here. 

https://github.com/phstc/shoryuken/wiki/Middleware

File: app/jobs/start_certification_job.rb [hunk 3]
```diff
@@ -1,5 +1,5 @@
 class StartCertificationJob < ActiveJob::Base
-  queue_as :default
+  queue_as :high_priority
```
Comment (sunil-sadasivan) [inline app/jobs/start_certification_job.rb]: Turning on the start certification job to use the high priority sqs queue.

**Comment by sunil-sadasivan** [comment]
@joofsh made some changes:

- turn on monitoring middleware for shoryuken
- updated all middleware to support shoryuken ([see docs](https://github.com/phstc/shoryuken/wiki/Middleware))
- adjusted middleware tests
- prefix queues with `caseflow_`
- turned on start_certification_job.  (this works well in local testing with sqs.  If you'd like to test this, run the shoryuken worker and adjust `app/model/certification.rb` to run a `StartCertificationJob.perform_later`)

File: app/jobs/middleware/job_prometheus_metric_middleware.rb [hunk 4]
```diff
@@ -1,17 +1,17 @@
 class JobPrometheusMetricMiddleware
-  def call(_worker, msg, _queue)
-    name = msg["args"][0]["job_class"]
+  def call(_worker, _queue, _msg, body)
+    job_class = body["job_class"]
```
Comment (joofsh) [inline app/jobs/middleware/job_prometheus_metric_middleware.rb]: Nice variable rename, much clearer 👍

File: config/initializers/shoryuken.rb [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+require "#{Rails.root}/app/jobs/middleware/job_request_store_middleware"
+
+# set up default exponential backoff parameters
+ActiveJob::QueueAdapters::ShoryukenAdapter::JobWrapper
+  .shoryuken_options(retry_intervals: [3.seconds, 30.seconds, 5.minutes, 30.minutes, 2.hours, 5.hours])
+
+Shoryuken.configure_server do |config|
+
+  # register all shoryuken middleware
+  config.server_middleware do |chain|
```
Comment (joofsh) [inline config/initializers/shoryuken.rb]: 🙌

**Pull request merged**